src/Lib.P2P/Routing/ContentRouter.cs

```diff
@@ -0,0 +1,144 @@
+        /// </summary>
+        /// <value>
+        ///   Defaults to 24 hours (1 day).
+        /// </value>
+        public TimeSpan ProviderTtl { get; set; } = TimeSpan.FromHours(24);
+
+        /// <summary>
+        ///    Adds the <see cref="Cid"/> and <see cref="Peer"/> to the content routing system.
+        /// </summary>
+        /// <param name="cid">
+        ///   The ID of some content that the <paramref name="provider"/> contains.
+        /// </param>
+        /// <param name="provider">
+        ///   The peer ID that contains the <paramref name="cid"/>.
+        /// </param>
+        public void Add(Cid cid, MultiHash provider) { Add(cid, provider, DateTime.Now); }
+
+        /// <summary>
+        ///   Adds the <see cref="Cid"/> and <see cref="Peer"/> to the content 
+        ///   routing system at the specified <see cref="DateTime"/>.
+        /// </summary>
+        /// <param name="cid">
+        ///   The ID of some content that the <paramref name="provider"/> contains.
+        /// </param>
+        /// <param name="provider">
+        ///   The peer ID that contains the <paramref name="cid"/>.
+        /// </param>
+        /// <param name="now">
+        ///   The local time that the <paramref name="provider"/> started to provide
+        ///   the <paramref name="cid"/>.
+        /// </param>
+        public void Add(Cid cid, MultiHash provider, DateTime now)
+        {
+            var pi = new ProviderInfo
+            {
+                Expiry = now + ProviderTtl,
+                PeerId = provider
+            };
+
+            _content.AddOrUpdate(
+                Key(cid),
+                (key) => new List<ProviderInfo> {pi},
+                (key, providers) =>
+                {
+                    var existing = providers
+                       .FirstOrDefault(p => p.PeerId == provider);
+                    if (existing != null)
+                    {
+                        existing.Expiry = pi.Expiry;
+                    }
+                    else
+                    {
+                        providers.Add(pi);
+                    }
+                    
+                    return providers;
+                });
+        }
+
+        /// <summary>
+        ///   Gets the providers for the <see cref="Cid"/>.
+        /// </summary>
+        /// <param name="cid">
+        ///   The ID of some content.
+        /// </param>
+        /// <returns>
+        ///   A sequence of peer IDs (providers) that contain the <paramref name="cid"/>.
+        /// </returns>
+        public IEnumerable<MultiHash> Get(Cid cid)
+        {
+            if (!_content.TryGetValue(Key(cid), out var providers))
+            {
+                return Enumerable.Empty<MultiHash>();
+            }
+
+            return providers
+               .Where(p => DateTime.Now < p.Expiry)
+               .Select(p => p.PeerId);
+        }
+
+        /// <inheritdoc />
+        public void Dispose() { }
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)